rotkehlchen/db/evmtx.py

```diff
@@ -154,6 +155,67 @@ def get_evm_internal_transactions(
 
         return transactions
 
+    def assert_tx_data_is_pulled(
+            self,
+            cursor: 'DBCursor',
+            evm_inquirer: 'EvmNodeInquirer',
+            tx_hash: 'EVMTxHash',
+            relevant_address: Optional['ChecksumEvmAddress'],
+    ) -> None:
+        """Makes sure that the receipt of the given transaction is in the database.
+        If not, pulls it and stores it in the database.
+
+        May raise:
+        - RemoteError if there is a problem querying the data source.
+        """
+        tx_receipt_data = cursor.execute(
+            'SELECT receipts.tx_id FROM evm_transactions AS transactions '
+            'INNER JOIN evmtx_receipts AS receipts ON transactions.identifier = receipts.tx_id '
+            'WHERE transactions.tx_hash = ?',
+            (tx_hash,),
+        ).fetchone()
+
+        if tx_receipt_data is not None:
+            return  # all good, tx receipt is in the database
+        
+        transaction, raw_receipt_data = evm_inquirer.get_transaction_by_hash(tx_hash)
+        # with self.db.conn.write_ctx() as write_cursor:
+        self.add_evm_transactions(
+            write_cursor=cursor,
+            evm_transactions=[transaction],
+            relevant_address=relevant_address,
+        )
+        self.add_receipt_data(
+            write_cursor=cursor,
+            chain_id=evm_inquirer.chain_id,
+            data=raw_receipt_data,
+            )
+
+    def get_or_create_transaction(
+            self,
+            cursor: 'DBCursor',
```

Comment:
If this is a write cursor then specify it in the name. We rely on the name being like this to know what type of cursor it is
```suggestion
            write_cursor: 'DBCursor',
```